src/ReleaseHistory.md

```diff
@@ -1,5 +1,10 @@
 # SARIF Package Release History (SDK, Driver, Converters, and Multitool)
 
+## 2.5.0.beta-1
+* BUGFIX: Loosen Newtonsoft.JSON minimum version requirement to 6.0.8 (for .NET framework) or 9.0.1 (for all other compilations) for Sarif.Sdk. Sarif.Converts requires 8.0.1, minimally, for .NET framework compilations.
```

Comment:
>BUGFIX [](http://example.com/codeflow?start=2&length=6)

I think both upgrade and downgrade Newtonsoft.JSON is a BREAKING change,
if we want to downgrade,
A user using old version of Sarif SDK, which was 12.0.3 or 10.0.3
(ignore the reverted 13.0.1 change)
Depending on his configuration he may be actually using Newtonsoft.JSON 6.0.8, after the dependent bot/manually bumps the Sarif SDK package.
1. he may have code that have issue with Newtonsoft.JSON 6.0.8
2. he may have pipeline with security scan got pipeline blocked with security issues between 6.0.8 and 12.0.3

Reply:
OK, not sure I understand this. If the only dependency on Newtonsoft comes from the SDK assembly, we're ok because only the SDK functionality is using it.

If there is another dependency on Newtonsoft, then that dependency *must* bet at least what it was before, say 12 or 13.0. If it isn't, the build won't work.

And so, when we drop the version, that already existing dependency version will still exist and SARIF will *still* be updated to the more recent version.

Reply:
If the only dependency on Newtonsoft comes from the SDK assembly  ---- my understanding is if so, unless the project set explicitly otherwise, we control the Newtonsoft version, and by using our sdk, if there is code interacting with the input and output json or json class we produce, it will be code that use Newtonsoft dll.

to repo, create an empty c# project, only add SARIF SDK package. 
Newtonsoft.Json v13 is added with the current published version.

At this point, if his code use V13 feature and works on V13, he checks it in.
We downgrade,
his build pipeline will get V6, it may break.

Reply:
OK, so how did 'his code use V13 features' without adding an explicit reference to V13 in that separate code project? You have to directly reference Newtonsoft in order to consume the V13 API. Am I still misunderstanding? Can you try to create a repro/example for me to look at?

Reply:
I mean when you ref a package, all dependency of the package will be added to the project, you can start using all functions in all of the dependency dlls. You do not need to manually add any version of Newtonsoft , if you add ref to SARIF SDK.

![image](https://user-images.githubusercontent.com/81775155/182515896-d30634b0-0bc1-440c-a8ae-a131e1084188.png)

to repo, create an empty c# project, only add SARIF SDK package, 

![image](https://user-images.githubusercontent.com/81775155/182516046-10b0e2b7-6f73-4c3d-a291-e8f8f19e0c91.png)

Reply:
And then if look at the .csproj it only has:
`<PackageReference Include="Sarif.Sdk" Version="2.4.16" />`

![image](https://user-images.githubusercontent.com/81775155/182518398-d49d372f-32af-4cae-926b-383f5e01934f.png)